src/types/datetime.rs

```diff
@@ -504,13 +529,54 @@ pub fn timezone_from_offset<'a>(py: Python<'a>, offset: &PyDelta) -> PyResult<&'
 /// Bindings for `datetime.timedelta`
 #[repr(transparent)]
 pub struct PyDelta(PyAny);
+#[cfg(not(Py_LIMITED_API))]
 pyobject_native_type!(
     PyDelta,
     crate::ffi::PyDateTime_Delta,
     |py| ensure_datetime_api(py).DeltaType,
     #module=Some("datetime"),
     #checkfunction=PyDelta_Check
 );
+// TODO: convert to a macro
```

Comment:
I am not sure whether we should try to have a wrapper `PyDelta` if only the limited API is available, i.e. does add little beyond `PyAny`. Wasn't the aim here to support the chrono conversions? If so, I think it would be easier (and more consistent with other conversions, c.f. e.g. `mod num_complex`) to have separate code paths to do the conversion, but not trying to provide a `mod datetime` that internally just calls the Python API.

Reply:
Yes, that's a good point. Let's close this PR and just implement fallback in `chrono` conversions

Reply:
Thank you!

Reply:
(This is probably a result of me not being very specific in https://github.com/PyO3/pyo3/issues/3633#issuecomment-1846844225)